src/configurePrim.tsx

```diff
@@ -1,23 +1,42 @@
-import React, { Component, useContext } from 'react'
-import { StyleProp, StyleSheet, TextStyle, ViewStyle } from 'react-native'
 import hoistNonReactStatics from 'hoist-non-react-statics'
+import React, { useContext } from 'react'
+import { TextStyle, ViewStyle, StyleSheet, StyleProp } from 'react-native'
 import { useMemoOne } from 'use-memo-one'
 
-// a string-value enum such as this one:
-//   enum CustomScreenSize {
-//     small = 'small'
-//     medium = 'medium'
-//     large = 'large'
-//     extraLarge = 'extraLarge'
-//   }
-// conforms to the following `StringValueEnum` type. this is usefule
-// for creating generic type constraints for such a type.
-type StringValueEnum = {
-  [id: string]: string
+type Children = {
+  children?: React.ReactNode
 }
 
 type CreateStyleSheet = typeof StyleSheet.create
 
+type ColorsConstraint = Record<string, ViewStyle['backgroundColor']>
+type ScreenSizeConstraint = Record<string, string>
+type SpacingConstraint = Record<string, ViewStyle['width']>
+type BorderRadiusConstraint = Record<string, ViewStyle['borderRadius']>
+type BorderWidthConstraint = Record<string, ViewStyle['borderWidth']>
+type FontSizeConstraint = Record<string, TextStyle['fontSize']>
+type FontWeightConstraint = Record<string, TextStyle['fontWeight']>
```

Comment:
nit: a helper type could could cut down on the (already minimal) duplication here.

```typescript
type StyleConstraint<Value> = Record<string, Value> 

type BorderRadiusConstraint = StyleConstraint<ViewStyle['borderRadius']>
```